ehr-connectivity/epic-fhirr4-administration-api-service/service.bal

```diff
@@ -0,0 +1,286 @@
+        r4:Bundle|error fhirInteractionResult = executeFhirInteraction("Patient", fhirContext, (), (), uscore:USCorePatientProfile).ensureType(r4:Bundle);
+        if fhirInteractionResult is error {
+            return r4:createFHIRError("Error occurred while executing the Patient search interaction.", r4:CODE_SEVERITY_ERROR,
+                r4:TRANSIENT_EXCEPTION, cause = fhirInteractionResult);
+        }
+        return fhirInteractionResult;
+    }
+
+    // Create a new resource with a server assigned id.
+    isolated resource function post fhir/r4/Patient(r4:FHIRContext fhirContext, uscore:USCorePatientProfile payload) returns @http:Payload {mediaType: ["application/fhir+json"]} Patient|r4:FHIRError {
+        Patient|error fhirInteractionResult = executeFhirInteraction("Patient", fhirContext, (), payload, uscore:USCorePatientProfile).ensureType(Patient);
+        if fhirInteractionResult is error {
+            return r4:createFHIRError("Error occurred while executing the Patient create interaction.", r4:CODE_SEVERITY_ERROR,
+                r4:TRANSIENT_EXCEPTION, cause = fhirInteractionResult);
+        }
+        return fhirInteractionResult;
+    }
+}
+
+service / on new fhirr4:Listener(9091, organizationApiConfig) {
+
+    // Read the current state of the resource.
+    isolated resource function get fhir/r4/Organization/[string id](r4:FHIRContext fhirContext) returns @http:Payload {mediaType: ["application/fhir+json"]} Organization|r4:FHIRError {
+        Organization|error fhirInteractionResult = executeFhirInteraction("Organization", fhirContext, id, (), uscore:USCoreOrganizationProfile).ensureType(Organization);
+        if fhirInteractionResult is error {
+            return r4:createFHIRError("Error occurred while executing the Organization read interaction.", r4:CODE_SEVERITY_ERROR,
+                r4:TRANSIENT_EXCEPTION, cause = fhirInteractionResult);
+        }
+        return fhirInteractionResult;
+    }
+
+    // Search the resource type based on some filter criteria.
+    isolated resource function get fhir/r4/Organization(r4:FHIRContext fhirContext) returns @http:Payload {mediaType: ["application/fhir+json"]} r4:Bundle|r4:FHIRError {
+        r4:Bundle|error fhirInteractionResult = executeFhirInteraction("Organization", fhirContext, (), (), uscore:USCoreOrganizationProfile).ensureType(r4:Bundle);
+        if fhirInteractionResult is error {
+            return r4:createFHIRError("Error occurred while executing the Organization search interaction.", r4:CODE_SEVERITY_ERROR,
+                r4:TRANSIENT_EXCEPTION, cause = fhirInteractionResult);
+        }
+        return fhirInteractionResult;
+    }
+
+    // Create a new resource with a server assigned id.
+    isolated resource function post fhir/r4/Organization(r4:FHIRContext fhirContext, uscore:USCoreOrganizationProfile payload) returns @http:Payload {mediaType: ["application/fhir+json"]} Organization|r4:FHIRError {
```

Comment:
Are we expecting only the USCore resources here or developer can send resources from custom packages?

isolated resource function post fhir/r4/Organization(r4:FHIRContext fhirContext, **uscore:USCoreOrganizationProfile** payload)

Reply:
It should be us core resources for the Epic EHR. This is added like this due to a issue on the fhir service type accepting the union types